crates/cheatnet/tests/starknet/cheat_fork.rs

```diff
@@ -0,0 +1,183 @@
+use crate::common::felt_selector_from_name;
+use crate::common::state::{create_cheatnet_state, create_fork_cached_state};
+use cairo_felt::Felt252;
+use cheatnet::rpc::{call_contract, CallContractOutput};
+use conversions::StarknetConversions;
+use num_bigint::BigUint;
+use starknet_api::core::ContractAddress;
+use std::str::FromStr;
+use test_case::test_case;
+
+const CAIRO0_TESTER_ADDRESS: &str =
+    "1825832089891106126806210124294467331434544162488231781791271899226056323189";
+
+#[test_case("return_caller_address"; "when common call")]
+#[test_case("return_proxied_caller_address"; "when library call")]
+fn prank_cairo0_contract(selector: &str) {
+    let mut cached_fork_state = create_fork_cached_state();
```

Comment:
What if somebody executes a library call from the test code on a cairo 0 class hash? Will library call code be cheated as well?

Reply:
Yup, added a test